io/src/pcd_io.cpp

```diff
@@ -971,7 +971,7 @@ pcl::PCDWriter::generateHeaderBinary (const pcl::PCLPointCloud2 &cloud,
          "\nFIELDS";
 
   // Compute the total size of the fields
-  unsigned int fsize = 0;
+  index_t fsize = 0;
```

Comment:
Don't switch to signed here. Cast `cloud.point_step` instead.

Reply:
Index_t is just what the user selects - so it could be signed/unsigned. Or do you want it to specifically be uindex_t?

Reply:
I'd suggest using `uindex_t` only when it makes sense to keep things unsigned. Rest of the times, we should be casting to `index_t`. Here, the size is obviously unsigned so I'd cast my vote with Sergio